src/golang/config/config.go

```diff
@@ -6,30 +6,78 @@ import (
 	"path"
 
 	"github.com/aqueducthq/aqueduct/lib/collections/shared"
-	log "github.com/sirupsen/logrus"
+	"github.com/dropbox/godropbox/errors"
 	"gopkg.in/yaml.v2"
 )
 
-type ServerConfiguration struct {
-	AqPath             string                `yaml:"aqPath" json:"aq_path"`
-	EncryptionKey      string                `yaml:"encryptionKey" json:"encryption_key"`
+var (
+	// globalConfigPath is set during Init
+	globalConfigPath string
+	// globalConfig is initialized during Init and updated as changes are made to the fields
+	globalConfig *serverConfiguration
+)
+
+type serverConfiguration struct {
+	AqPath             string                `yaml:"aqPath"`
+	EncryptionKey      string                `yaml:"encryptionKey"`
 	RetentionJobPeriod string                `yaml:"retentionJobPeriod"`
 	ApiKey             string                `yaml:"apiKey"`
 	StorageConfig      *shared.StorageConfig `yaml:"storageConfig"`
 }
 
-func ParseServerConfiguration(confPath string) *ServerConfiguration {
-	bts, err := ioutil.ReadFile(confPath)
+// AqueductPath is the filepath to the Aqueduct installation.
+func AqueductPath() string {
+	return globalConfig.AqPath
+}
+
+// EncryptionKey is used for encrypting objects stored in the Aqueduct vault.
+func EncryptionKey() string {
+	return globalConfig.EncryptionKey
+}
+
+// RetentionJobPeriod defines how long to wait before garbage collecting workflow runs.
+func RetentionJobPeriod() string {
+	return globalConfig.RetentionJobPeriod
+}
+
+// APIKey returns the API key the user must use when issuing requests.
+func APIKey() string {
+	return globalConfig.ApiKey
+}
+
+// Storage returns the storage layer config.
```

Comment:
This is a snapshot/copy of the storage config right? Can we make that explicit in the name (eg. CopyStorageConfig() or SnapshotStorageConfig()). I ask because everywhere else we seem to use pointers to the same config, and this place is the exception, and an important one.

Reply:
I wanted to return a copy to make it explicit that the only way to modify the storage config is via the `UpdateStorage` function in this pkg. Do you think it's sufficient to just state this in the comments?